xrootd/xrdproto/protocol.go

```diff
@@ -52,6 +52,25 @@ func (err ServerError) Error() string {
 	return fmt.Sprintf("xrootd: error %d: %s", err.Code, err.Message)
 }
 
+// MarshalXrd implements Marshaler.
+func (o ServerError) MarshalXrd(wBuffer *xrdenc.WBuffer) error {
+	wBuffer.WriteI32(int32(o.Code))
+	wBuffer.WriteBytes([]byte(o.Message))
+	wBuffer.WriteBytes([]byte("\x00"))
+	return nil
+}
+
+// UnmarshalXrd implements Unmarshaler.
+func (o *ServerError) UnmarshalXrd(rBuffer *xrdenc.RBuffer) error {
+	o.Code = ServerErrorCode(rBuffer.ReadI32())
+	data := rBuffer.Bytes()
+	if len(data) == 0 {
+		return errors.New("xrootd: an error message was not provided in the server response")
```

Comment:
`s/xrootd: an error message was not provided in the server response/xrootd: missing error message in server response/`